Add merge commits information to DORA change lead time documentation (#25092)
DataDog/documentation: content/en/dora_metrics/data_collected/_index.md

These metrics are only computed when the source of the repository metadata is GitHub, and there must be a pull request (PR) associated with a commit, if any. A commit is associated with a PR if the commit is first introduced to the target branch when merging that PR. If a commit does not have an associated PR, only `dora.time_to_deploy` and `dora.deploy_time` metrics are available.

**Notes:**

- These metrics are emitted for every commit and not per deployment.

> - Make sure to familiarize with the [DORA metrics limitations][12].

Author: @gturbat and I chatted offline and think it's better to just add a link to the limitations from here to avoid the two identical blocks of text getting out of sync. @brett0000FF what do you think?

> - There are several edge cases depending on the way the commits were introduced to the deployment, view the [limitations][12].

Reviewer: @GabrielAnca Agreed, that would make things better from a maintenance perspective. We could create a reusable snippet as well, but in this instance linking seems simpler. Happy to help either way though. 😄

> - There are several edge cases depending on the way the commits were introduced to the deployment, view the [limitations][12].

Author: Nice, I didn't know about reusable snippets! But in any case I'm happy with the link here for now too! 😄

These metrics are only computed when the source of the repository metadata is GitHub, and there must be a pull request (PR) associated with a commit, if any. A commit is associated with a PR if the commit is first introduced to the target branch when merging that PR. If a commit does not have an associated PR, only `dora.time_to_deploy` and `dora.deploy_time` metrics are available.

**Notes:**

- These metrics are emitted for every commit and not per deployment.

> - Commits that are not meaningful changes (for example merge commits) are ignored.

Reviewer: ```suggestion
- Commits without meaningful changes (for example, merge commits) are ignored.
```

> - For rebased feature branches, *change lead time* calculations consider the new commits created during the rebase, not the original commits.

Reviewer 2: ```suggestion
- For rebased branches, *change lead time* calculations consider the new commits created during the rebase, not the original commits.
```
It would also be true if you do a rebase on the main/deployment branches (nobody would do this, but still)

> - If commits on a feature branch are rebased, the commits considered for the change lead time are the new commits created with the rebase and not the original commits.

Reviewer: ```suggestion
- For rebased commits on a feature branch, *change lead time* calculations consider the new commits created during the rebase, not the original commits.
```

Author: I wanted to double check this change: `rebased` here is used as a verb, ie: "when you rebase a commit" and it feels like the change switches it to an adjective. 

But I think the usage of commit is wrong there, we don't rebase commits, we rebase branches (my bad!). What do you think about this?

```suggestion
- For rebased feature branches, *change lead time* calculations consider the new commits created during the rebase, not the original commits.
- ```

Reviewer: Ah! All good. Yeah, I agree with your proposed edits. 🚀

- When using "Squash and merge" for pull requests:
  - For GitHub: Metrics are emitted for the original commits.

>   - For other providers: Metrics are emitted for the new commit added to the target branch.

Reviewer 2: ```suggestion
  - For other git providers: Metrics are emitted for the new commit added to the target branch.
```
Just to explicit it has nothing to do with the CI provider

> - "Squash and merge", when a pull request is squashed:
>   - For GitHub: the metrics are emitted for the original commits.
>   - For other providers: the metrics are emitted for the new commit added to the target branch.

Reviewer: ```suggestion
- When using "Squash and merge" for pull requests:
  - For GitHub: Metrics are emitted for the original commits.
  - For other providers: Metrics are emitted for the new commit added to the target branch.
```

These metrics are only computed when the source of the repository metadata is GitHub, and there must be a pull request (PR) associated with a commit, if any. A commit is associated with a PR if the commit is first introduced to the target branch when merging that PR. If a commit does not have an associated PR, only `dora.time_to_deploy` and `dora.deploy_time` metrics are available.

**Notes:**

- These metrics are emitted for every commit and not per deployment.
- Commits without meaningful changes (for example, merge commits) are ignored.

> - For rebased feature branches, *change lead time* calculations consider the new commits created during the rebase, not the original commits.
> - ```
> - When using "Squash and merge" for pull requests:

Author: Oops I made a typo here

```suggestion
- For rebased feature branches, *change lead time* calculations consider the new commits created during the rebase, not the original commits.
- When using "Squash and merge" for pull requests:
```
